src/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -752,9 +755,12 @@ private final static boolean isVCFStream(final InputStream stream, final String
         return new LazyGenotypesContext.LazyData(genotypes, header.getSampleNamesInOrder(), header.getSampleNameToOffset());
     }
 
-    private final String[] INT_DECODE_ARRAY = new String[10000];
+    private final int LARGE_SIZE = 10000;
+    private final String[] INT_DECODE_ARRAY = new String[2*LARGE_SIZE];
```

Comment:
Agreed, it shouldn't specifically mention PLs. Other genotype fields may cause the same issue.